pkg/clusteragent/autoscaling/workload/controller.go

```diff
@@ -174,7 +175,12 @@ func (c *Controller) syncPodAutoscaler(ctx context.Context, key, ns, name string
 		if podAutoscaler != nil {
 			// If we don't have an instance locally, we create it. Deletion is handled through setting the `Deleted` flag
 			log.Debugf("Creating internal PodAutoscaler: %s from Kubernetes object", key)
-			c.store.UnlockSet(key, model.NewPodAutoscalerInternal(podAutoscaler), c.ID)
+			podAutoscalerInternal := model.NewPodAutoscalerInternal(podAutoscaler)
+			c.store.UnlockSet(key, podAutoscalerInternal, c.ID)
+			c.AutoscalerHeap.InsertIntoHeap(autoscaling.TimestampKey{
```

Comment:
Instead of having to track changes to `Store`, could be it registered as an observer?